modules/oe_theme_content_project/src/Plugin/ExtraField/Display/PercentageExtraField.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_content_project\Plugin\ExtraField\Display;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Cache\CacheableMetadata;
+
+/**
+ * Extra field displaying eu budget percentage on projects.
+ *
+ * @ExtraFieldDisplay(
+ *   id = "oe_theme_content_project_percentage",
+ *   label = @Translation("Percentage"),
+ *   bundles = {
+ *     "node.oe_project",
+ *   },
+ *   visible = true
+ * )
+ */
+class PercentageExtraField extends ProjectExtraFieldBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLabel() {
+    // Add space as label so the value aligns with the other fields.
+    return ' ';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    // Compute budget percentage field value.
+    $budget = $entity->get('oe_project_budget')->value;
+    $budget_eu = $entity->get('oe_project_budget_eu')->value;
+    $percentage = $this->getPercentage((float) $budget, (float) $budget_eu);
+    $build = [
+      '#plain_text' => $this->t("@percentage% of the overall budget", ["@percentage" => $percentage]),
+    ];
+    $cache = new CacheableMetadata();
+    $cache->addCacheableDependency($entity);
+    $cache->applyTo($build);
```

Comment:
I think we can skip this, as the field's entity is always a render cache dependency of its fields, right @upchuk ?

Reply:
Removed but please @upchuk confirm that we dont need it. Thank you